### PR TITLE
Add a networking customization section in the HOWTO

### LINKED ISSUE #324: Document network customization HOWTO
The assumption is that networking is managed through NetworkManager, which is true for all our shipped images.

The configuration at present could be done by creating NetworkManager configuration files via cloud-init.
Creating the files itself is easy, how to generate NetworkManager config files not.
Document the whole process in an HOWTO section.

Should include VLAN ~and Bond interfaces~ in the example.

### PR DESCRIPTION
**Pull request by fgiudici**
Fixes #324

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -82,30 +82,40 @@ const sidebars = {
...
+      "label": "How to",
+      "items": [
+        "custom-images",
+        "elemental_behind_proxy",
+        "hostname",
+        {
+          "type": "category",
+          "collapsible": true,
+          "collapsed": true,
+          "label": "Costumize networking",
```
Comment (juadk) [inline sidebars.js:97]:
```suggestion
          "label": "Customize networking",
```

File: docs/networking.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+<head>
+  <link rel="canonical" href="https://elemental.docs.rancher.com/networking"/>
+</head>
+
+
+## Network configuration with Elemental
+
+Elemental cloud-config support do not include declarative networking at the moment.
```
Comment (juadk) [inline docs/networking.md:13]:
```suggestion
Elemental cloud-config support does not include declarative networking at the moment.
```

File: docs/networking.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  ipv4.method manual \
+  ipv4.addresses 192.168.1.2/24 \
+  ipv4.gateway 192.168.1.1 \
+  ipv4.dns 192.168.1.1
+```
+After running the above command, you will find your keyfile at
+`/etc/NetworkManager/system-connections/fixed-ip.nmconnection`.
+
+### Elemental cloud-config
+The NetworkManager connection profile keyfiles can be injected it in a [MachineRegistration](machineregistration-reference) using the `write_files` module.
```
Comment (juadk) [inline docs/networking.md:58]:
```suggestion
The NetworkManager connection profile keyfiles can be injected in a [MachineRegistration](machineregistration-reference) using the `write_files` module.
```

File: docs/networking.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+The `content`` can be either plain text or base64 encoded (`encoding: b64`).
```
Comment (juadk) [inline docs/networking.md:60]: Wrong formatting I guess, I checked the preview and it looks bad:
image

Comment (fgiudici) [reply docs/networking.md:60]: good catch! 
```suggestion
The `content` can be either plain text or base64 encoded (`encoding: b64`).
```

**Review by juadk — APPROVED** [review]
LGTM

**Pull request merged**